src/main/scala/org/ergoplatform/network/ErgoNodeViewSynchronizer.scala

```diff
@@ -56,31 +59,124 @@ class ErgoNodeViewSynchronizer(networkControllerRef: ActorRef,
     context.system.scheduler.scheduleAtFixedRate(toDownloadCheckInterval, toDownloadCheckInterval, self, CheckModifiersToDownload)
   }
 
+  private def downloadRequired(historyReader: ErgoHistory)(id: ModifierId): Boolean =
+    deliveryTracker.status(id, Seq(historyReader)) == ModifiersStatus.Unknown
```

Comment:
@pragmaxim Seq constructor is quite slow comparing to Array, which can safely be used here, because the parameter is of type Seq, and the array will be wrapped into WrappedArray.
Please read this document and try to use those recommendations https://github.com/ScorexFoundation/sigmastate-interpreter/blob/develop/docs/perf-style-guide.md